Sia/Buffers/BucketBuffer.cs

```diff
@@ -70,33 +84,25 @@ public bool IsAllocated(int index)
     }
 
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    public ref T GetRef(int index)
```

Comment:
The core of the GetRef is to throw error if need. But we need keep data safe in production